core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -1105,18 +1105,18 @@ protected void cleanUp() {
     }
 
     private List<PrepareJob> buildSubnetworkRemovalJobs() {
-        List<FlagEncoder> encoders = encodingManager.fetchEdgeEncoders();
         List<PrepareJob> jobs = new ArrayList<>();
-        for (FlagEncoder encoder : encoders) {
-            // for encoders with turn costs we do an edge-based subnetwork removal, because they *might* be used with
-            // a profile with turn_costs=true
-            if (encoder.supportsTurnCosts()) {
-                // u-turn costs are zero as we only want to make sure the graph is fully connected assuming finite
-                // u-turn costs
+        for (Profile profile : profilesByName.values()) {
+            FlagEncoder encoder = encodingManager.getEncoder(profile.getVehicle());
+            Weighting weighting = createWeighting(profile, new PMap());
+            if (profile.isTurnCosts()) {
+                // u-turn costs are zero as we only want to make sure the graph is fully connected assuming finite u-turn costs
                 TurnCostProvider turnCostProvider = new DefaultTurnCostProvider(encoder, ghStorage.getTurnCostStorage(), 0);
-                jobs.add(new PrepareJob(encoder.toString(), encoder.getAccessEnc(), turnCostProvider));
+                jobs.add(new PrepareJob(profile.getName(), encodingManager.getBooleanEncodedValue(InSubnetwork.key(profile.getName())),
+                        encoder.getAccessEnc(), weighting, turnCostProvider));
             } else {
-                jobs.add(new PrepareJob(encoder.toString(), encoder.getAccessEnc(), TurnCostProvider.NO_TURN_COST_PROVIDER));
+                jobs.add(new PrepareJob(profile.getName(), encodingManager.getBooleanEncodedValue(InSubnetwork.key(profile.getName())),
```

Comment:
> There should be no need to pass the access encoded value anymore here.

Only the subnetwork EV is passed

> passing in the weighting anyway it would be nice to get rid of the turn cost provider argument.

Will do!

Reply:
> Only the subnetwork EV is passed

ups, I see :)

Reply:
Did you mean something like in the latest commit?

Reply:
Yes this looks better now.